src/agent0/chainsync/dashboard/build_ticker.py

```diff
@@ -2,34 +2,116 @@
 
 import pandas as pd
 
-from .usernames import map_addresses
+from .usernames import abbreviate_address, map_addresses
 
 
-def build_ticker(ticker_data: pd.DataFrame, user_map: pd.DataFrame) -> pd.DataFrame:
-    """Show recent trades.
+def build_pool_ticker(
+    trade_events: pd.DataFrame, user_map: pd.DataFrame, block_to_timestamp: pd.DataFrame
+) -> pd.DataFrame:
+    """Show recent trades wrt a pool.
 
     Arguments
     ---------
-    ticker_data: pd.DataFrame
-        The dataframe resulting from get_transactions.
+    trade_events: pd.DataFrame
+        The dataframe resulting from `get_trade_events`.
     user_map: pd.DataFrame
-        A dataframe with 5 columns (address, abbr_address, username, user, format_name).
-        This is the output of :meth:`chainsync.dashboard.build_user_mapping`.
+        A dataframe containing the mapping of wallet addresses to usernames.
+    block_to_timestamp: pd.DataFrame
+        A dataframe containing the mapping of block number to timestamp.
 
     Returns
     -------
     pd.DataFrame
-        The filtered transaction data based on what we want to view in the ticker.
+        The filtered transaction data based on what we want to view in the ticker for a specific pool.
     """
     # Gather other information from other tables
-    mapped_addrs = map_addresses(ticker_data["wallet_address"], user_map)
+    mapped_addrs = map_addresses(trade_events["wallet_address"], user_map)
+
+    trade_events = trade_events.copy()
+    trade_events["username"] = mapped_addrs["username"]
+
+    # Look up block to timestamp
+    trade_events = trade_events.merge(block_to_timestamp, how="left", on="block_number")
+
+    rename_dict = {
+        "timestamp": "Timestamp",
+        "block_number": "Block Number",
+        "username": "User",
+        "wallet_address": "Wallet",
+        "event_type": "Trade",
+        "token_id": "Token",
+        "token_delta": "Token Change",
+        "base_delta": "Base Change",
+        "as_base": "As Base",
+    }
+    trade_events = trade_events[list(rename_dict.keys())].rename(columns=rename_dict)
 
-    ticker_data = ticker_data.copy()
-    ticker_data = ticker_data.drop("id", axis=1)
-    ticker_data.insert(2, "username", mapped_addrs["username"])
-    ticker_data.columns = ["Block Number", "Timestamp", "User", "Wallet", "Method", "Token Deltas"]
     # Shorten wallet address string
-    ticker_data["Wallet"] = mapped_addrs["abbr_address"]
-    # Drop rows with nonexistant wallets
-    ticker_data = ticker_data.dropna(axis=0, subset="Wallet")
-    return ticker_data
+    trade_events["Wallet"] = mapped_addrs["abbr_address"]
+
+    # Sort latest first
```

Comment:
is this sort actually happening? or is the comment noting that it is already the case?